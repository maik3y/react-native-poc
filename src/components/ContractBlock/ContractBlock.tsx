import React from 'react'
import { Text as DefaultText, View as DefaultView } from 'react-native'
import { styled } from 'nativewind'

import { ContractBlockRow } from './ContractBlockRow/ContractBlockRow'

interface ContractBlockProps {
  number: string
  name: string
  reference: string
  location: {
    street: string
    city: string
    number: string
  }
}

export default function ContractBlock({ data }: { data: ContractBlockProps }) {
  const View = styled(DefaultView)
  const Text = styled(DefaultText)

  return (
    <View tw="flex w-full rounded-md border dark:border-white mb-2 overflow-hidden">
      <View tw="flex w-full bg-blue p-2 bg-purple-600">
        <Text tw="text-white font-bold">Contract {data.number}</Text>
      </View>
      <ContractBlockRow label={'naam'} value={data.name} />
      <ContractBlockRow
        label={'project'}
        value={`${data.location.street} | ${data.location.city}`}
      />
      <ContractBlockRow label={'Jouw referentie'} value={data.reference} />
      <ContractBlockRow
        label={'Locatie'}
        value={`${data.location.street} ${data.location.number}, ${data.location.city}`}
      />
    </View>
  )
}
