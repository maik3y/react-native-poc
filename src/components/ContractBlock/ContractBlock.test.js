import React from 'react'
import renderer from 'react-test-renderer'
import ContractBlock from './ContractBlock'

const mockData = {
  number: '123456',
  name: 'test',
  reference: 'test reference',
  location: {
    street: 'street',
    city: 'city',
    number: '13'
  }
}

test('renders correctly', () => {
  const tree = renderer.create(<ContractBlock data={mockData} />).toJSON()
  expect(tree).toMatchSnapshot()
})
