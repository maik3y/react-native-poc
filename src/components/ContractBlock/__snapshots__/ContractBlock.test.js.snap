// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "display": "flex",
        },
        Object {
          "width": "100%",
        },
        Object {
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
        },
        Object {
          "borderBottomWidth": 1,
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
        },
        Object {
          "marginBottom": 8,
        },
        Object {
          "overflow": "hidden",
        },
      ],
    ]
  }
>
  <View
    style={
      Array [
        Array [
          Object {
            "display": "flex",
          },
          Object {
            "width": "100%",
          },
          Object {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          },
          Object {
            "backgroundColor": "#9333ea",
          },
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Array [
            Object {
              "color": "#fff",
            },
            Object {
              "fontWeight": "700",
            },
          ],
        ]
      }
    >
      Contract 
      123456
    </Text>
  </View>
  <View
    style={
      Array [
        Array [
          Object {
            "display": "flex",
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "flexWrap": "wrap",
          },
          Object {
            "width": "100%",
          },
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Array [
            Object {
              "display": "flex",
            },
            Object {
              "width": "33.333333%",
            },
            Object {
              "paddingBottom": 2,
              "paddingLeft": 2,
              "paddingRight": 2,
              "paddingTop": 2,
            },
          ],
        ]
      }
    >
      <Text>
        naam
      </Text>
    </View>
    <View
      style={
        Array [
          Array [
            Object {
              "display": "flex",
            },
            Object {
              "width": "66.666667%",
            },
            Object {
              "paddingBottom": 2,
              "paddingLeft": 2,
              "paddingRight": 2,
              "paddingTop": 2,
            },
          ],
        ]
      }
    >
      <Text>
        test
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Array [
          Object {
            "display": "flex",
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "flexWrap": "wrap",
          },
          Object {
            "width": "100%",
          },
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Array [
            Object {
              "display": "flex",
            },
            Object {
              "width": "33.333333%",
            },
            Object {
              "paddingBottom": 2,
              "paddingLeft": 2,
              "paddingRight": 2,
              "paddingTop": 2,
            },
          ],
        ]
      }
    >
      <Text>
        project
      </Text>
    </View>
    <View
      style={
        Array [
          Array [
            Object {
              "display": "flex",
            },
            Object {
              "width": "66.666667%",
            },
            Object {
              "paddingBottom": 2,
              "paddingLeft": 2,
              "paddingRight": 2,
              "paddingTop": 2,
            },
          ],
        ]
      }
    >
      <Text>
        street | city
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Array [
          Object {
            "display": "flex",
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "flexWrap": "wrap",
          },
          Object {
            "width": "100%",
          },
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Array [
            Object {
              "display": "flex",
            },
            Object {
              "width": "33.333333%",
            },
            Object {
              "paddingBottom": 2,
              "paddingLeft": 2,
              "paddingRight": 2,
              "paddingTop": 2,
            },
          ],
        ]
      }
    >
      <Text>
        Jouw referentie
      </Text>
    </View>
    <View
      style={
        Array [
          Array [
            Object {
              "display": "flex",
            },
            Object {
              "width": "66.666667%",
            },
            Object {
              "paddingBottom": 2,
              "paddingLeft": 2,
              "paddingRight": 2,
              "paddingTop": 2,
            },
          ],
        ]
      }
    >
      <Text>
        test reference
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Array [
          Object {
            "display": "flex",
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "flexWrap": "wrap",
          },
          Object {
            "width": "100%",
          },
          Object {
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Array [
            Object {
              "display": "flex",
            },
            Object {
              "width": "33.333333%",
            },
            Object {
              "paddingBottom": 2,
              "paddingLeft": 2,
              "paddingRight": 2,
              "paddingTop": 2,
            },
          ],
        ]
      }
    >
      <Text>
        Locatie
      </Text>
    </View>
    <View
      style={
        Array [
          Array [
            Object {
              "display": "flex",
            },
            Object {
              "width": "66.666667%",
            },
            Object {
              "paddingBottom": 2,
              "paddingLeft": 2,
              "paddingRight": 2,
              "paddingTop": 2,
            },
          ],
        ]
      }
    >
      <Text>
        street 13, city
      </Text>
    </View>
  </View>
</View>
`;
